addons/schedule/common/src/main/java/org/commonjava/indy/schedule/ScheduleDBUtil.java

```diff
@@ -0,0 +1,36 @@
+package org.commonjava.indy.schedule;
+
+import org.commonjava.indy.schedule.conf.ScheduleDBConfig;
+
+public class ScheduleDBUtil
+{
+
+    public static final String TABLE_SCHEDULE = "schedule";
+
+    public static String getSchemaCreateKeyspace( ScheduleDBConfig config, String keyspace )
+    {
+        return "CREATE KEYSPACE IF NOT EXISTS " + keyspace
+                        + " WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor':"
+                        + config.getReplicationFactor() + "};";
+    }
+
+    public static String getSchemaCreateTableSchedule( String keyspace )
+    {
+        return "CREATE TABLE IF NOT EXISTS " + keyspace + "." + TABLE_SCHEDULE + " ("
+                        + "jobtype varchar,"
+                        + "jobname varchar,"
+                        + "storekey varchar,"
+                        + "scheduletime timestamp,"
+                        + "lifespan bigint,"
+                        + "expired boolean,"
+                        + "ttl bigint,"
```

Comment:
What is difference between ttl and lifespan?

Reply:
ttl here as a column we set TTL, 
`"UPDATE " + keyspace +  "." + ScheduleDBUtil.TABLE_SCHEDULE + " USING TTL ? SET ttl=? WHERE storekey = ? and  jobname = ?"`
when the entry is created, we set ttl on the column "ttl", then we can query it out by the following way cassandra provide (select ttl(<column> from ...)),
`"SELECT storekey, jobtype, jobname, scheduletime, lifespan, expired, ttl(ttl) as ttl FROM " + keyspace + "." + ScheduleDBUtil.TABLE_SCHEDULE `
 to see if the entry is expired ( the value of the function ttl(<column>) is 0, and the value of the column itself is updated to null ).  While the column "lifespan" will be kept that we can audit later.

Reply:
Do you mean cassandra do not remove the entry when getting ttl ? when is it get removed?

Reply:
No, cassandra can remove it, but it does not trigger anything like what ispn does, so we need to query it out and trigger the event if needed.

Reply:
There are some comments for the TTL, https://projects.engineering.redhat.com/browse/MMENG-713.
Welcome thoughts. :)

Reply:
I see. thanks. As far as I can see, this design seems unusual. The frequency of querying will be critical. Will this involve full table cross partition query? e.g. if query per second or some short time period, this might not work once the table grow big. if query less often, the latency will be big.

Reply:
Yeah, for this case, it needs to scan the full table to check the expired ones. But we don't need to do that per second, I was thinking to make the query at least per min, even 5 mins. Do we have the case that need to do this more frequently ? But sure I agree with you that this will cost if we do the query across the partitions.